packages/@sanity/types/src/schema/definition/type/common.ts

```diff
@@ -0,0 +1,49 @@
+import {ComponentType, ReactElement, ReactNode} from 'react'
+import {ConditionalProperty} from '../../types'
+import {ObjectOptions} from './object'
+
+export type FieldsetDefinition = {
+  name: string
+  title?: string
+  description?: string
+  hidden?: ConditionalProperty
+  readOnly?: ConditionalProperty
+  options?: ObjectOptions
+}
+
+export type FieldGroupDefinition = {
+  name: string
+  title?: string
+  icon?: ComponentType | ReactNode
+  default?: boolean
+}
+
+export interface BaseDefinitionOptions {
```

Comment:
Name this `BaseDefinition`?

Reply:
I agree, renaming to BaseSchemaDefinition